src/shared/theme/validate.ts

```diff
@@ -73,7 +73,11 @@ export function getContexts(preset: ThemePreset) {
   const contexts: string[] = [];
 
   for (const theme of themes) {
-    contexts.push(...Object.keys(theme.contexts));
+    Object.keys(theme.contexts).forEach((contextName) => {
+      if (contexts.indexOf(contextName) === -1) {
```

Comment:
nit: this could be optimised with a Set (which would be _O(1)_ vs the current _O(n)_):

Something like:

```typescript
const contexts = new Set<string>();

for (const theme of themes) {
  Object.keys(theme.contexts).forEach((contextName) => {
    contexts.add(contextName);
  });
}

return Array.from(contexts);
```

Reply:
Good point! Addressed

Reply:
Actually, I set it back to the previous implementation because I got a build error
```
src/shared/theme/validate.ts(81,16): error TS2550: Property 'from' does not exist on type 'ArrayConstructor'. Do you need to change your target library? Try changing the 'lib' compiler option to 'es2015' or later.
```

The number of contexts is limited, so I don't think this will be a performance issue.

Reply:
Yes, it was a nitpick. Alternatively, I think using the spread operator might have worked, but this is non-blocking, so I've approved!